Sources/SwiftJWT/JWTDecoder.swift

```diff
@@ -154,36 +160,49 @@ private struct _JWTKeyedDecodingContainer<Key: CodingKey>: KeyedDecodingContaine
     // A reference to the Decoder the container is inside
     let decoder: _JWTDecoder
     
-    /// A reference to the container we're reading from.
-    private let container: [String : Data]
+    var header: Data
+    
+    var claims: Data
     
     var codingPath: [CodingKey]
     
-    public var allKeys: [Key] {
+    public var allKeys: [Key]
+    {
         #if swift(>=4.1)
-        return self.container.keys.compactMap { Key(stringValue: $0) }
+            return ["header", "claims"].compactMap { Key(stringValue: $0) }
         #else
-        return self.container.keys.flatMap { Key(stringValue: $0) }
+            return ["header", "claims"].flatMap { Key(stringValue: $0) }
         #endif
     }
     
-    fileprivate init(decoder: _JWTDecoder, wrapping container: [String : Data]) {
+    fileprivate init(decoder: _JWTDecoder, header: Data, claims: Data) {
         self.decoder = decoder
-        self.container = container
+        self.header = header
+        self.claims = claims
         self.codingPath = decoder.codingPath
     }
     
     public func contains(_ key: Key) -> Bool {
-        return self.container[key.stringValue] != nil
+        return key.stringValue == "header" || key.stringValue == "claims"
```

Comment:
Would `return allKeys.contains(key)` work?

Reply:
Key is not equatable so we can't use the contains function.